### DISCUSSION TITLE
Home Assistant Dashboard view

### DISCUSSION THREAD

**Discussion by Pseudomizer** [RobertD502/home-assistant-flair#49] [Ideas]
Hi,

first of all THANK you for developing this plugin. Installation was a breeze and you can't say this about many plugins. 

With that said, I have now added to each room on my Dashboard the vent or vents for each room. For the status of the vent I am getting a diagonal vent for close and open. This is your way of showing the status meaning if it is close, it is grayed out and you can only click the open. For somebody who doesn't know this, you have look at it first and then compare the two and look at the white one to understand if it is open or closed. 

My suggestion would be to make this 3 fields. One for status which dynamically updates and then two buttons or keep the current arrows.  Here are 3 suggestions from ideal to easiest to implement.

Ideal: (Under action it only shows the option available)
Status: **CLOSED**  Action: **Open** 
or 
Status: **OPEN**  Action: **Close**

Alternative: (I cannot show white and gray in this editor but assume the bold is white and the other one is grayed out)
Status: **CLOSED**  Action: **Open**  Close
or 
Status: **OPEN**  Action: Open  **Close**

Easier change:
Status: **CLOSED**  Action: **↗** ↘
or 
Status: **OPEN**  Action: ↗  **↘**

**Comment by RobertD502** [comment]
Hi @Pseudomizer! Thanks for using the component. Unfortunately, this isn't something that I can change - The frontend and how entities are displayed in the UI is handled by Home Assistant. Components don't have the ability to customize the open/close action and what they look like based on the open/closed state - there is a standard based on entity types. Although it may seem weird to those that are unfamiliar with Home Assistant, it is something that they'll learn and it translates to any other integration that is using similar entities.

Just to explain a little bit about cover entities and when arrows are greyed out or not:

1. If the vent is completely closed, then only the Up arrow (open the vent) action is available....pressing this sets the vent to completely open.

![image](https://user-images.githubusercontent.com/52541649/218292759-fe14b44e-248a-475f-87f5-e8a17720808b.png)

2. If the vent is completely open, then only the down arrow (close the vent) action is available....pressing this sets the vent to completely closed.

![image](https://user-images.githubusercontent.com/52541649/218292801-dfec2657-7c97-4137-b971-56380dd3d600.png)

3. Now here is where it may be confusing, but it makes sense. Flair vents have a third option besides fully open or fully closed - 50% open. If you have your vent 50% open, then both open and closing actions will be available. Logically it makes sense since, at 50% open, the available ways that you can act on the vent are to open it more (to 100% by pressing the up arrow) or closing it more (to 0% open by pressing the down arrow).

![image](https://user-images.githubusercontent.com/52541649/218292860-398f5cf7-a734-4984-a786-dbdda12b1756.png)

If at any point you want to get the vent to 50% open, you have to click on the entity to set the "tilt position".



Another example I can give you is the native Home Assistant Soma integration (smart blinds):

![image](https://user-images.githubusercontent.com/52541649/218292954-ff30603e-d0ab-4291-902d-5f28c50f3ceb.png)

Blinds are also presented as cover entities. The greying out/highlighting of the up and down arrows, based on if the blinds are open or closed, is automatically handled by Home Assistant and can't be changed from the integration level. Disregard the square action button - this is an option to pause an open or close action that is in progress (something that doesn't relate to Flair vents as there is no such command).

**Reply by RobertD502** [reply]
To add to this: If you are looking for ways to customize how your vents look on your dashboard, I'd look around HACS for custom lovelace cards that handle cover entities.

**Reply by RobertD502** [reply]
Check out [mushroom cards](https://github.com/piitaya/lovelace-mushroom).

Looks like the [cover card](https://github.com/piitaya/lovelace-mushroom/blob/main/docs/cards/cover.md) will let you set the state as secondary info while maintaining the arrow buttons.

**Comment by Pseudomizer** [comment]
Thank you so much. Such insight is really helpful. Your plugin will gain popularity as they start selling Flair vents again. I would humbly suggest to add this information on your front page. Will be helpful for other users as well.
